### DISCUSSION TITLE
Intended use of `[Options] output` ?

### DISCUSSION THREAD

**Discussion by CodeMan99** [Orange-OpenSource/hurl#2532] [Q&A] [answered]
# Hello!

I'm pretty new to `hurl` in general.

## Context

As a proof of concept, I wanted to _bootstrap download_ hurl with a hurlfile.

```hurl
GET https://github.com/Orange-OpenSource/hurl/releases/latest
Accept: application/json
[Options]
location: true
HTTP 200
[Captures]
version: jsonpath "$.tag_name"


GET https://github.com/Orange-OpenSource/hurl/releases/download/{{version}}/hurl_{{version}}_amd64.deb
[Options]
location: true
output: hurl_{{version}}_amd64.deb
```

Running this file with `hurl get_hurl.hurl` does work. However, because it is the last request, the deb file is also output on stdout. The first time I did this I assumed that nothing was saved due to seeing the binary barf all over my screen.

## Question

Basically, is it intended that the output is **both** the specified file and stdout?

## Other Discussions

https://github.com/Orange-OpenSource/hurl/discussions/1323

**Comment by jcamiel** [comment, marked as answer]
Hi,

It's good to have newcomers because we can see what's need to be improved, in the documentation and in the way we present Hurl.

The intended way to behave for Hurl is: _By default, Hurl executes all HTTP requests and outputs the response body of the last HTTP call._ (see [here](https://hurl.dev/docs/manual.html#description)).

So, given this Hurl file:

```hurl
GET https://example.org/foo.txt

GET https://example.org/bar.txt
```

And run like this:

```shell
$ hurl sample.hurl
```

Hurl will output the content of `bar.txt` on standard output. This is the designed (expected?) behavior. We've done this to be kind of like curl, and to get a unique response in workflows that require multiple steps.

You can save the output of the last response like this:

```shell
$ hurl sample.hurl > out.txt
```

Or

```shell
$ hurl --output out.txt sample.hurl
```

Note that we have a current bug for binary responses (see https://github.com/Orange-OpenSource/hurl/issues/2306): like curl, for binary responses, we should not output the last response to standard output.

We have also a way to specify options on a particular request with `[Options]` sections. We can use it if we want intermediate responses:

```hurl
GET https://example.org/foo.txt
[Options]
output: out1.txt

GET https://example.org/bar.txt

GET https://example.org/baz.txt
[Options]
output: out2.txt
```

And run with this:

```shell
$ hurl --no-output sample.hurl
```

I've specified `--no-output` because I've written in the Hurl file where I want my last response to be saved.

With Hurl 4.3.0 (not released yet), you can use `-` to indicate that you want a specific response on stdout:

```hurl
# This will go on standard output
GET https://example.org/foo.txt
[Options]
output: -

GET https://example.org/bar.txt

GET https://example.org/baz.txt
[Options]
output: out2.txt
```

I hope it's more clear!

**Reply by CodeMan99** [reply]
Thanks for explaining. Is there an in-file option to turn off the stdout response of the last request? Can I for example do this?

```hurl
GET https://example.com/foo.txt

GET https://example.com/bar.txt
[Options]
output: bar.txt
no-output: true
```

Note that I have not yet attempted this.
